### PR TITLE
New hunting and detection queries for suspicious host activity

### PR DESCRIPTION
**Pull request by petebryan**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - New content:
    Fodhelper UAC Bypass
    Discord Downloads from command line
    Crash Dump disablement
    All queries include an ASIM version

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - Yes

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

File: Hunting Queries/SecurityEvent/Crashdumpdisabledonhost.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+requiredDataConnectors:
+  - connectorId: SecurityEvents
+    dataTypes:
+      - SecurityEvents
+tactics:
+  - DefenseEvasion
+relevantTechniques:
+  - T1070
+query: |
+  SecurityEvent
```
Comment (shainw) [inline Hunting Queries/SecurityEvent/Crashdumpdisabledonhost.yaml:14]: Why are we duplicating this when we already have imRegistry above?

File: Hunting Queries/SecurityEvent/Discorddownloadinvokedfromcmdline.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+tactics:
+  - Execution
+  - CommandAndControl
+  - Exfiltration
+relevantTechniques:
+  - T1204
+  - T1102
+  - T1567
+query: |
+    SecurityEvent
```
Comment (shainw) [inline Hunting Queries/SecurityEvent/Discorddownloadinvokedfromcmdline.yaml:18]: Why are we duplicating this when we have imProcess for this above?

**Review by shainw — CHANGES_REQUESTED** [review]

File: Detections/SecurityEvent/PotentialFodhelperUACBypass.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+queryFrequency: 1h
+queryPeriod: 2h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1548.002
+query: |
+  SecurityEvent
```
[the remark this answers is not in the document]
Comment (petebryan) [reply Detections/SecurityEvent/PotentialFodhelperUACBypass.yaml:19]: Neither imRegistry or imProcess and built in parsers. There will be a large number of workspaces that can't currently leverage them directly so providing both options for greatest reach and impact.

File: Hunting Queries/SecurityEvent/Crashdumpdisabledonhost.yaml (hunk 1 above)
Comment (petebryan) [reply Hunting Queries/SecurityEvent/Crashdumpdisabledonhost.yaml:14]: Neither imRegistry or imProcess and built in parsers. There will be a large number of workspaces that can't currently leverage them directly so providing both options for greatest reach and impact.

File: Hunting Queries/SecurityEvent/Discorddownloadinvokedfromcmdline.yaml (hunk 2 above)
Comment (petebryan) [reply Hunting Queries/SecurityEvent/Discorddownloadinvokedfromcmdline.yaml:18]: Neither imRegistry or imProcess are built in parsers. There will be a large number of workspaces that can't currently leverage them directly so providing both options for greatest reach and impact.

**Review by shainw — APPROVED** [review]

File: Detections/MultipleDataSources/PotentialFodhelperUACBypassASIM.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  - T1548.002
+query: |
+  imRegistry
+    | where EventType in ("RegistryValueSet", "RegistryKeyCreated")
+    | where RegistryKey has "Software\\Classes\\ms-settings\\shell\\open\\command"
+    | extend timekey = bin(TimeGenerated, 1h)
+    | join (imProcess
+    | where Process endswith "fodhelper.exe"
+    | where ParentProcessName endswith "cmd.exe" or ParentProcessName endswith "powershell.exe" or ParentProcessName endswith "powershell_ise.exe"
+    | extend timekey = bin(TimeGenerated, 1h)) on TimeKey, DvcId
```
Comment (oshezaf) [inline Detections/MultipleDataSources/PotentialFodhelperUACBypassASIM.yaml:23]: @petebryan : I suspect Dvc would be better here. DvcId tends to be specific to a source (Sysmon, MDE, SecurityEvent). Dvc is the best identified we have for the device, most often hostname, where you have better chance of correlating between different sources. 

What do you think?

File: Hunting Queries/ASimRegistry/CrashdumpdisabledonhostASIM.yaml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+description: |
+  'This detection looks the prevention of crash dumps being created. This can be used to limit reporting by malware, look for suspicious processes setting this registry key.'
+requiredDataConnectors: []
+tactics:
+  - DefenseEvasion
+relevantTechniques:
+  - T1070
+query: |
+    imRegistry
+    | where RegistryKey == "SYSTEM\\CurrentControlSet\\Control\\CrashControl"
```
Comment (oshezaf) [inline Hunting Queries/ASimRegistry/CrashdumpdisabledonhostASIM.yaml:12]: imRegistry uses the MDE style registry names. See [here](https://docs.microsoft.com/en-us/azure/sentinel/registry-event-normalization-schema#root-keys)

File: Detections/MultipleDataSources/PotentialFodhelperUACBypassASIM.yaml (hunk 4 above)
Comment (oshezaf) [inline Detections/MultipleDataSources/PotentialFodhelperUACBypassASIM.yaml:23]: @petebryan : I suspect Dvc would be better here. DvcId tends to be specific to a source (Sysmon, MDE, SecurityEvent). Dvc is the best identified we have for the device, most often hostname, where you have better chance of correlating between different sources. 

What do you think?

**Review by oshezaf — COMMENTED** [review]
@petebryan : one of the comments does require a fix, even though we merged.

**Pull request merged**